rasa/core/training/loading.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os
+from typing import Text, Optional, Dict, List, Any
+
+import rasa.utils.io as io_utils
+from rasa.core.domain import Domain
+from rasa.core.interpreter import NaturalLanguageInterpreter, RegexInterpreter
+from rasa.core.training.structures import StoryStep
+
+logger = logging.getLogger(__name__)
+
+
+def _get_reader_class(filename: Text) -> Optional[Any]:
+
+    from rasa.core.training.story_reader import yaml_story_reader, markdown_story_reader
+
+    module = None
```

Comment:
That's a good question, I think we can generalize it more: `Do we want to fail fast, or try to load as much training data as we can?`

I don't have a strong opinion here. My thinking is:
- We have only two story readers, bot are in the same package. So if the user has issues with the installation, they will have to fix that first
- We anyway print the warning to the output, and in case the have a "custom" story reader that was not found, at least they will be able to train some model and see results

So IMO it's fine not to fail fast here and rather leave the user with at least some model if they are just experimenting.

Reply:
>  and in case the have a "custom" story reader 

How would they even use a custom story reader? They could do so through implementing a custom `TrainingDataImporter`  but in that case they would not even use this code here.

> We have only two story readers, bot are in the same package. So if the user has issues with the installation, they will have to fix that first

When there are issues with the installation, then no reader can be loaded, correct? So then the training wouldn't work anyway. 

> So IMO it's fine not to fail fast here and rather leave the user with at least some model if they are just experimenting.

The question is, what would be left to train with? And how likely is that case in the first place?
Arguments I see for against `load as much as we can`
- they might not see the warning and then might just end up confused
- we have to add extra code to handle `None` (especially since it should be just an an edge case)
- we are also failing early for stuff like `TrackerStore`

Reply:
Though about it more - I think you have a great point that users might not even notice the log message and get very confused. Let's throw then 👍